pkg/service/node.go

```diff
@@ -21,23 +21,60 @@ import (
 
 // NodeService implements the CSI Driver node service
 type NodeService struct {
-	infraClusterClient kubernetes.Clientset
-	kubevirtClient     kubevirt.Client
 	nodeID             string
+	deviceLister       deviceLister
+	fsMaker            fsMaker
+	fsMounter          mount.Interface
+	dirMaker           dirMaker
+}
+
+func NewNodeService(infraClusterClient kubevirt.Client, nodeId string) *NodeService {
+	return &NodeService{
+		nodeID:             nodeId,
+		deviceLister:       lsblkLister,
+		fsMaker:            fsMakerBase,
+		fsMounter:          mount.New(""),
+		dirMaker: func(path string, perm os.FileMode) error {
+			// MkdirAll returns nil if path already exists
+			return os.MkdirAll(path, perm)
+		},
+	}
 }
 
 var nodeCaps = []csi.NodeServiceCapability_RPC_Type{
 	csi.NodeServiceCapability_RPC_STAGE_UNSTAGE_VOLUME,
 }
 
+type callableFunc func() ([]byte, error)
```

Comment:
I'm going to simplify this a bit, and maybe make the function type interfaces type